core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java

```diff
@@ -89,14 +95,37 @@ public Object getValue() {
         });
   }
 
+  @VisibleForTesting
+  protected void addMultiValueAggregator(MultiValueMetricsAggregator aggregator) {
+    mMultiValueMetricsAggregatorRegistry.add(aggregator);
+  }
+
+  private void updateMultiValueMetrics() {
+    for (MultiValueMetricsAggregator aggregator : mMultiValueMetricsAggregatorRegistry) {
+      Map<MetricsFilter, Set<Metric>> metrics = new HashMap<>();
+      for (MetricsFilter filter : aggregator.getFilters()) {
+        metrics.put(filter, mMetricsStore.getMetricsByInstanceTypeAndName(filter.getInstanceType(),
+            filter.getName()));
+      }
+      for (Entry<String, Long> entry : aggregator.updateValues(metrics).entrySet()) {
+        MetricsSystem.registerGaugeIfAbsent(entry.getKey(), new Gauge<Object>() {
```

Comment:
Why do we use `Gauge` here, is it possible that we would want something other than a Gauge?

Reply:
According to the definition, a gauge is an instantaneous measurement of a value. This suits for the aggregation value, which is a view over the metrics from the distributed instances.